pkg/relayer/config/types.go

```diff
@@ -0,0 +1,132 @@
+package config
+
+import "net/url"
+
+// YAMLRelayMinerConfig is the structure used to unmarshal the RelayMiner config file
+// TODO_DOCUMENT(@red-0ne): Add proper README documentation for yaml config files.
+type YAMLRelayMinerConfig struct {
+	SigningKeyName string                         `yaml:"signing_key_name"`
+	SmtStorePath   string                         `yaml:"smt_store_path"`
+	Pocket         YAMLRelayMinerPocketConfig     `yaml:"pocket"`
+	Proxies        []YAMLRelayMinerProxyConfig    `yaml:"proxies"`
+	Suppliers      []YAMLRelayMinerSupplierConfig `yaml:"suppliers"`
+}
+
+// YAMLRelayMinerPocketConfig is the structure used to unmarshal the pocket
+// node URLs section of the RelayMiner config file
+type YAMLRelayMinerPocketConfig struct {
+	QueryNodeGRPCUrl string `yaml:"query_node_grpc_url"`
```

Comment:
We'll need to add inline comments here (not now) when we add the proper documentation as well.